pages/api/register.js

```diff
@@ -36,13 +37,21 @@ const handler = nc({
       // Encrypt user password
       const encryptedPassword = await bcrypt.hash(password, 10);
 
+      const uniqueString = await emailService.createUniqueString();
       // Create user in our database
-      const user = await userService.createUser(email, encryptedPassword, name, age);
+      const user = await userService.createUser(email, encryptedPassword, name, age, uniqueString);
+
+      await emailService.sendEmail(email, uniqueString);
 
       // Create token
-      const token = await tokenService.createToken(user.id, email);
+      // const token = await tokenService.createToken(user.id, email);
+
+      // return res.status(201).json({ message: 'user registered!', token });
 
-      return res.status(201).json({ message: 'user registered!', token });
+      return res.status(201).json({
+        message: `user ${user.id} registered! Please confirm your email to login`,
+        uniqueString: user.uniqueString,
```

Comment:
If you sent a secret string back to the client they will be able to call the `verify` endpoint without accessing their inbox.